hoomd/md/minimize/fire.py

```diff
@@ -139,49 +198,59 @@ def __init__(self,
 
         super().__init__(forces, constraints, methods, rigid)
 
         self._param_dict.update(
-            ParameterDict(
-                dt=float(dt),
-                aniso=OnlyFrom(['true', 'false', 'auto'],
-                               preprocess=_preprocess_aniso),
-                min_steps_adapt=OnlyTypes(int,
-                                          preprocess=_raise_if_nonpositive),
-                finc_dt=float,
-                fdec_dt=float,
-                alpha_start=float,
-                fdec_alpha=float,
-                force_tol=float,
-                angmom_tol=float,
-                energy_tol=float,
-                min_steps_conv=OnlyTypes(int, preprocess=_raise_if_nonpositive),
-                _defaults={
-                    "aniso": "auto",
-                    "min_steps_adapt": 5,
-                    "finc_dt": 1.1,
-                    "fdec_dt": 0.5,
-                    "alpha_start": 0.1,
-                    "fdec_alpha": 0.99,
-                    "force_tol": 0.1,
-                    "angmom_tol": 0.1,
-                    "energy_tol": 1e-5,
-                    "min_steps_conv": 10
-                }))
-
-    def _validate_methods(self):
-        """Ensure that only NVE/NPH integration methods are used."""
-        for method in self.methods:
-            method_name = method.__class__.__name__
-            if method_name not in ["NVE", "NPH"]:
-                raise RuntimeError("{} integration method not compatible with "
-                                   "FIRE energy minimizer".format(method_name))
+            ParameterDict(dt=float(dt),
+                          aniso=OnlyFrom(['true', 'false', 'auto'],
+                                         preprocess=_preprocess_aniso),
+                          min_steps_adapt=OnlyTypes(int,
+                                                    preprocess=positive_real),
+                          finc_dt=float,
+                          fdec_dt=float,
+                          alpha_start=float,
+                          fdec_alpha=float,
+                          force_tol=float,
+                          angmom_tol=float,
+                          energy_tol=float,
+                          min_steps_conv=OnlyTypes(int,
+                                                   preprocess=positive_real),
+                          _defaults={
+                              "aniso": "auto",
+                              "min_steps_adapt": min_steps_adapt,
+                              "finc_dt": finc_dt,
+                              "fdec_dt": fdec_dt,
+                              "alpha_start": alpha_start,
+                              "fdec_alpha": fdec_alpha,
+                              "force_tol": force_tol,
+                              "angmom_tol": angmom_tol,
+                              "energy_tol": energy_tol,
+                              "min_steps_conv": min_steps_conv
+                          }))
```

Comment:
This will bypass any check any validation.

Reply:
How do I get the aniso value to be checked upon construction, since it is an `OnlyTypes` object?

Reply:
You will need to construct the `ParameterDict` then update it with the passed in values. There is no way to validate on construction.